src/components/ListItems.jsx

```diff
@@ -0,0 +1,309 @@
+import React, { useState, useRef, useEffect } from "react";
+import { useDispatch } from "react-redux";
+
+// @material-ui import
+import IconButton from "@material-ui/core/IconButton";
+import DeleteIcon from "@material-ui/icons/Delete";
+import AddCircleRoundedIcon from "@material-ui/icons/AddCircleRounded";
+import RemoveCircleIcon from "@material-ui/icons/RemoveCircle";
+
+// actions import
+import { addItem, deleteItem } from "../actions/ItemActions";
+import {
+    addSelectedTopicsItem,
+    deleteSelectedTopicsItem,
+} from "../actions/selectedTopicsItemActions";
+
+import "../App.css";
+
+export const ListItems = (props) => {
+    // Initial setup
+    const [btnAddNewItemClicked, setBtnAddNewItemClicked] = useState(false);
+    const [inputEditedItemVisible, setInputEditedItemVisible] = useState(false);
+    const [inputNewItem, setInputNewItem] = useState("");
+    const [editedItem, setEditedItem] = useState(null);
+    const [editedItemIndex, setEditedItemIndex] = useState(null);
+    const [inputEditedItem, setInputEditedItem] = useState("");
+
+    const dispatch = useDispatch();
+
+    const handleDelete = (e) => {
+        const index = e.currentTarget.value;
+        const items = props.items;
+        const selectedItem = items[index];
+        const selectedItems = props.selectedItems || [];
+        const exist = selectedItems.includes(selectedItem);
+        let action = props.type === "topics" ? "DELETE_TOPIC" : "DELETE_PLAYER";
+
+        // delete item from selectedItems too
+        if (exist) {
+            const indexSelectedItem = selectedItems.indexOf(selectedItem);
+            dispatch(
+                deleteSelectedTopicsItem(
+                    items[indexSelectedItem],
+                    "DELETE_SELECTED_ITEM"
+                )
+            );
+        }
+        dispatch(
+            deleteItem(
+                items[index],
+                action
+            )
+        );
+    };
+
+    const handleClickAdd = (e) => {
+        setBtnAddNewItemClicked(true);
+        const previousSibling = e.currentTarget.previousElementSibling;
+
+        if (btnAddNewItemClicked) {
+            if (previousSibling.value === "") {
```

Comment:
please explain why you need to track which button has been clicked !
The if branch does not execute on click because state setters are asynchronous - how does this even work ?

Reply:
The first click to this button make appear an input, second click, add value (if not empty) to state and hide input. To do this i need to know if it's clicked or not.